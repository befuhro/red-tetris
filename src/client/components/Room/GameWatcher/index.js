import React from 'react';

import Square from "../Square";

const gameWatcherStyle = {
    justifySelf: 'end',
    width: '22vh',
    height: '70vh',
    margin: '12.8vh 0 0 0',
    padding: '2vh 2vh 0 2vh',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    backgroundColor: '#0b1d2d',
    borderColor: '#0b1d2d',
    borderWidth: '1vh 2vh 0 2vh',
    borderStyle: 'outset',
};

const spectrumStyle = {
    color: 'white'
};

const specBoardType = {
    width: '13vh',
    height: '20vh',
    display: 'flex',
    flexWrap: 'wrap',
    padding: '0',
    margin: '0',
};

const Spectrum = ({playerName, spectrum}) =>
    <div style={spectrumStyle}>
        {playerName}
        <div style={specBoardType}>
            {
                spectrum.map(line =>
                        line.map(bloc => bloc === 1 ?
                        <Square color={'realwhite'} size={{width: '1vh', height: '1vh', borderWidth: '0.1vh'}}/> :
                        <Square color={'white'} size={{width: '1vh', height: '1vh', borderWidth: '0.1vh'}}/>)
                )
            }
        </div>
    </div>;

export default ({players}) => {
    return (
        <div style={gameWatcherStyle}>
            <div>
                {
                    Object.values(players).map((player, index) => {
                            return (<Spectrum
                                key={index}
                                playerName={player.username}
                                spectrum={player.spectrum}
                            />)
                        }
                    )
                }
            </div>
        </div>);
}
