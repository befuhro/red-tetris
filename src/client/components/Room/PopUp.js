import React from 'react';
import { Link } from 'react-router-dom';

const backgroundPopUp = {
    backgroundColor: 'rgba(255, 255, 255, 0.5)',
    position: 'fixed',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
};

const popUp = {
    justifyContent: 'center',
    alignItems: 'center',
    position: 'absolute',
    display: 'flex',
    flexDirection: 'column',
    padding: '10px',
    backgroundColor: '#0b1d2d',
};

const errorFont = {
    color: '#61dafb',
    fontFamily: 'sans-serif',
    margin: '10px 20px 10px 20px',
};

const link = {
    fontFamily: 'sans-serif',
    backgroundColor: 'white',
    width: 'fit-content',
    padding: '5px',
    color: 'black',
    textDecoration: 'none',
    borderRadius: '5px',
};

function PopUp({errors}) {
    return (
        <div style={backgroundPopUp}>
            <div style={popUp}>
                {errors.map((error) => <p key={error.id} style={errorFont}>{error.message}</p>)}
                <Link style={link} to="/">Login Page</Link>
            </div>
        </div>
    )
}

export default PopUp;
