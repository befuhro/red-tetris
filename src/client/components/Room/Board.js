import React, { useEffect } from 'react';

import {BOTTOM} from "../../utils/direction";
import {  movePiece, changeInterval } from "../../actions/room";

import Square from './Square';
import logo from '../../assets/logo.svg';



const mainBlockStyle = {
    width: '35vh',
    height: '80vh',
    margin: '2.7vh 0 0 0',
    padding: '2vh 2vh 0 2vh',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    backgroundColor: '#0b1d2d',
    borderColor: '#0b1d2d',
    borderWidth: '1vh 2vh 0 2vh',
    borderStyle: 'outset',
};

const titleStyle = {
    fontFamily: 'Orbitron',
    fontWeight: '500',
    fontSize: '4vh',
    textAlign: 'center',
    color: '#e1301d'
};

const boardStyle = {
    width: '29vh',
    height: '60vh',
    display: 'flex',
    flexWrap: 'wrap',
    borderColor: '#0f1423',
    borderWidth: '2vh 2vh 1vh 2vh',
    borderRadius: '1vh 1vh 0 0',
    borderStyle: 'inset'
};

export default ({ board }) => {

    useEffect(() => {
    });


    return (
        <div style={mainBlockStyle} id={'env'}>
            <img src={logo} className='App-logo' alt='react-logo'/>
            <h1 style={titleStyle}>RED TETRIS</h1>
            <div style={boardStyle}>
                {
                    board.map((square, index) =>
                        <Square
                            key={index}
                            index={index}
                            color={square.color}
                        />)
                }
            </div>
        </div>
    )
}
