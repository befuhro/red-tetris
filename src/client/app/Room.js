import React, {useEffect} from 'react';
import {connect} from 'react-redux';
import WebFont from 'webfontloader';

import GameController from '../room/gameController/GameController';
import {env} from '../Store/Actions/Env';
import Board from '../room/game/Game';
// import GameWatcherComponent from './GameWatcherComponent';
import Footer from './Footer';
import {askPiece, listenerGame, movePiece, rotatePiece} from "../Store/Actions/Game";
import {BOTTOM, LEFT, RIGHT} from "../utils/direction";

WebFont.load({google: {families: ['Permanent Marker', 'Orbitron:black']}});


const appStyle = {
    height: '100%'
};

const gameStyle = {
    display: 'flex',
    justifyContent: 'space-around'
};

function handleKey(event, dispatch) {
    if (event.key === 'ArrowLeft') {
        dispatch(movePiece(LEFT));
    }
    if (event.key === 'ArrowRight') {
        dispatch(movePiece(RIGHT));
    }
    if (event.key === 'ArrowDown') {
        dispatch(movePiece(BOTTOM));
    }
    if (event.key === 'ArrowUp') {
        dispatch(rotatePiece());
    }
    if (event.key === ' ') {
        dispatch(askPiece());
    }
}


function connectToRoom(dispatch, hashPlayer, hashRoom, statePlayer, stateRoom) {
    if ((hashPlayer !== statePlayer || hashRoom !== stateRoom) &&
        (hashPlayer !== undefined && hashRoom !== undefined)) {
        dispatch(env(hashPlayer, hashRoom));
    }
}

function Room({match: {params}, dispatch, username, room, isConnected, isListening}) {
    useEffect(() => {
        connectToRoom(dispatch, params.player, params.room, username, room);
        if (!isListening) {
            dispatch(listenerGame());
            document.addEventListener('keydown', (event) => handleKey(event, dispatch));
        }
    });
    return (
        <div style={appStyle}>
            <div style={gameStyle}>
                {/*<GameWatcherComponent/>*/}
                <Board />
                <GameController />
            </div>
            <Footer />
        </div>
    )
}

const mapStateToProps = (state) => ({
    username: state.env.username,
    room: state.env.room,
    isConnected: state.env.isConnected,
    isListening: state.game.isListening,
});

const mapDispatchToProps = (dispatch) => ({
    dispatch: dispatch,
});

export default connect(mapStateToProps, mapDispatchToProps)(Room);