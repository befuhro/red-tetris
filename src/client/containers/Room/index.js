import {connect} from 'react-redux';

import Room from '../../components/Room';

const mapStateToProps = (state) => ({errors: state.room.errors, intervalMove: state.room.intervalMove});

const mapDispatchToProps = dispatch => ({ dispatch: dispatch });

export default connect(mapStateToProps, mapDispatchToProps)(Room);
