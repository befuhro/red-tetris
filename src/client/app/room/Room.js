import React, { useEffect } from 'react';
import {connect} from 'react-redux';
import WebFont from 'webfontloader';

import GameController from './controller/Controller';
// import { env } from '../../Store/Actions/Env';
import Board from './board/Board';
import {joinRoom} from "./roomActions";
import PopUp from './PopUp';
import Footer from "../Footer";
import {askPiece, movePiece, rotatePiece} from "./roomActions";
import {BOTTOM, LEFT, RIGHT} from "../../utils/direction";

WebFont.load({google: {families: ['Permanent Marker', 'Orbitron: black']}});

const gameStyle = {
    display: 'flex',
    justifyContent: 'space-around'
};

function handleKey(event, dispatch) {
    switch (event.key) {
        case 'ArrowLeft':
            dispatch(movePiece(LEFT));
            break;
        case 'ArrowRight':
            dispatch(movePiece(RIGHT));
            break;
        case 'ArrowDown':
            dispatch(movePiece(BOTTOM));
            break;
        case 'ArrowUp':
            dispatch(rotatePiece());
            break;
        case ' ':
            dispatch(askPiece());
            break;
    }
}

function Room({dispatch, errors, match: {params}}) {

    console.log(window.RTCPeerConnection);

    useEffect(() => {
        dispatch(joinRoom(params.player, params.room));
    }, []);

    return (
        <div onKeyDown={(e) => handleKey(e, dispatch)} tabIndex={0} >
            {errors === null ?
                <div style={gameStyle}>
                    {/*<GameWatcherComponent/>*/}
                    <Board/>
                    <GameController/>
                </div>
                :
                <PopUp errors={errors}/>
            }
            <Footer/>
        </div>
    )
}

const mapStateToProps = (state) => ({
    errors: state.room.errors
});

const mapDispatchToProps = dispatch => ({ dispatch: dispatch });

export default connect(mapStateToProps, mapDispatchToProps)(Room);