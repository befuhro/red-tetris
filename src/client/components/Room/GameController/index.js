import React from 'react';

import PieceView from './PieceView';
import { startParty } from "../../../actions/room/roomActions";

const gameControllerStyle = {
    justifySelf: 'end',
    width: '22vh',
    height: '70vh',
    margin: '12.8vh 0 0 0',
    padding: '2vh 2vh 0 2vh',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    backgroundColor: '#0b1d2d',
    borderColor: '#0b1d2d',
    borderWidth: '1vh 2vh 0 2vh',
    borderStyle: 'outset',
};

const buttonStyle = {
    padding: '1vh',
    fontFamily: 'Orbitron',
    fontWeight: '500',
    fontSize: '2vh',
    textAlign: 'center',
    color: '#e1301d',
    backgroundColor: '#5a6978',
    borderColor: '#0f1423',
    borderWidth: '1vh 1vh 1vh 1vh',
    borderRadius: '1vh 1vh 1vh 1vh',
    borderStyle: 'outset'
};

function StartButton({ isRoomLeader, gameIsStarted, dispatch }) {
    return (
        <div>
            {isRoomLeader && gameIsStarted !== true ?
                <button style={buttonStyle} onClick={() => dispatch(startParty())}>Start Game</button>
                :
                null
            }
        </div>
    );
}

export default ({ isRoomLeader, gameIsStarted, firstPiece, secondPiece, thirdPiece, dispatch}) => {
    return (
        <div style={gameControllerStyle}>
            <div>
                <PieceView piece={thirdPiece}/>
                <PieceView piece={secondPiece}/>
                <PieceView piece={firstPiece}/>
            </div>
            <StartButton isRoomLeader={isRoomLeader} gameIsStarted={gameIsStarted} dispatch={ dispatch }/>
        </div>);
}
