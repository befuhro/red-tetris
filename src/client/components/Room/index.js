import React, { useEffect } from 'react';
import WebFont from 'webfontloader';

import GameController from '../../containers/Room/GameController';
import GameWatcher from "../../containers/Room/GameWatcher";

import { joinRoom, askPiece, movePiece, rotatePiece } from "../../actions/room";
import { BOTTOM, LEFT, RIGHT } from "../../utils/direction";

import Board from '../../containers/Room/Board';
import PopUp from './PopUp';
import Footer from "../shared/Footer";

WebFont.load({google: {families: ['Permanent Marker', 'Orbitron: black']}});

const gameStyle = {
    display: 'flex',
    justifyContent: 'space-around'
};

function handleKey(dispatch) {
    return function(e) {
        switch (e.key) {
            case 'ArrowLeft':
                dispatch(movePiece(LEFT));
                break;
            case 'ArrowRight':
                dispatch(movePiece(RIGHT));
                break;
            case 'ArrowDown':
                dispatch(movePiece(BOTTOM));
                break;
            case 'ArrowUp':
                dispatch(rotatePiece());
                break;
            case ' ':
                dispatch(askPiece());
                break;
        }
    }
}

export default ({dispatch, intervalMove, errors, match: {params}}) => {

    useEffect(() => {
        dispatch(joinRoom(params.player, params.room));
        const curriedEvent = handleKey(dispatch);
        const interval = setInterval(() => {
            dispatch(movePiece(BOTTOM));
        }, intervalMove);
        document.addEventListener('keydown', curriedEvent);

        return () => {
            document.removeEventListener('keydown', curriedEvent);
            clearInterval(interval);
        }
    });

    return (
        <div onKeyDown={(e) => handleKey(e, dispatch)} tabIndex={0} >
            {errors === null ?
                <div style={gameStyle}>
                    <GameWatcher/>
                    <Board/>
                    <GameController/>
                </div>
                :
                <PopUp errors={errors}/>
            }
            <Footer/>
        </div>
    )
}
