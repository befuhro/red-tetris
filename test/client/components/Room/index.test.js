import * as React from 'react';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {MemoryRouter} from "react-router-dom";


enzyme.configure({adapter: new Adapter()});

import Room, {handleKey} from '../../../../src/client/components/Room';

describe('<Room>', () => {

    describe('basic test', () => {
        let component;

        beforeAll(() => {
            component = enzyme.mount(
                <MemoryRouter initialEntries={['/']}>
                    <Room
                        dispatch={() => {
                        }}
                        errors={[]}
                        match={{
                            params: {player: 'befuhro', room: '101'}
                        }}
                    />
                </MemoryRouter>
            )
        });
        it('basic render', () => expect(component.html()).toMatchSnapshot());
    });

    describe('test handle key', () => {
        it('test key', () => {
            const dispatchTest = object => action => expect(action).toMatchObject(object);
            handleKey(dispatchTest({type: 'MOVE_PIECE', direction: 'LEFT'}))({key: 'ArrowLeft'});
            handleKey(dispatchTest({type: 'MOVE_PIECE', direction: 'RIGHT'}))({key: 'ArrowRight'});
            handleKey(dispatchTest({type: 'MOVE_PIECE', direction: 'BOTTOM'}))({key: 'ArrowDown'});
            handleKey(dispatchTest({type: 'MOVE_PIECE', direction: 'LOWEST'}))({key: ' '});
            handleKey(dispatchTest({type: 'ROTATE_PIECE'}))({key: 'ArrowUp'});
            handleKey(null)({key: 'test'});
        });
    });
});
