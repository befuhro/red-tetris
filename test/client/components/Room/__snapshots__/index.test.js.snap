// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Room> basic test basic render 1`] = `"<div id=\\"main-game\\" tabindex=\\"0\\"><div style=\\"background-color: rgba(255, 255, 255, 0.5); position: fixed; display: flex; justify-content: center; align-items: center; top: 0px; left: 0px; right: 0px; bottom: 0px;\\"><div style=\\"justify-content: center; align-items: center; position: absolute; display: flex; flex-direction: column; padding: 10px; background-color: rgb(11, 29, 45);\\"><a style=\\"font-family: sans-serif; background-color: white; padding: 5px; color: black; text-decoration: none; border-radius: 5px;\\" href=\\"/\\">Login Page</a></div></div><div style=\\"display: flex; justify-content: center; align-items: center; position: absolute; bottom: 0px; width: 100%; height: 14.24vh; color: rgb(97, 218, 251);\\"><p style=\\"font-family: Permanent Marker; font-size: 3vh; font-weight: 300; color: rgb(97, 218, 251); margin-right: 2vh;\\">befuhro &amp; eruaud</p><img src=\\"\\" style=\\"height: 10vh; margin-left: 2vh;\\" alt=\\"le-101-logo\\"></div></div>"`;
